### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Mhdez221993**
# Project requirements

* Create a class Corrector with a method `correct_name`
* Set an instance of `Corrector` in `Person` on initialization
* In `Person` add a method `validate_name` that uses the `Corrector` instance to validate and save the corrected name

# Thank you for taking the time to review my project 👍

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    @parent_permission = parent_permission
+    @corrector = Corrector.new
+  end
+
+  def can_use_services?
+    is_of_age or @parent_permission
+  end
+
+  def validate_name
+    @corrector.correct_name(@name)
```
Comment (Mihndim2020) [inline person.rb]: - [x] This method should use the corrector instance to validate and save the corrected name like, `@name = @corrector.correct_name(@name)`. Please kindly fix that.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @Mhdez221993, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Mhdez221993) [reply person.rb]: Done!

**Review by bimbolabuari — APPROVED** [review]
# Project Approved 🚀

Hi @Mhdez221993 🥇,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**